rust/src/index/vector/pq.rs

```diff
@@ -257,14 +276,68 @@ impl VectorIndex for PQIndex {
         let row_ids =
             read_fixed_stride_array(reader, &DataType::UInt64, row_id_offset, length, ..).await?;
 
+        // Precollecting everything to a hash map requires async closure. This way we avoid that
+        let mut deletion_vec_cache: HashMap<u64, DeletionVector> = HashMap::new();
```

Comment:
If we need to duplicate this for other indices, should we encapsulate this into it's own struct? I think that also let's us not bring the manifest and object store down into here.

Reply:
this cache is only valid within each look up. Let me add deletion handling in diskANN. What you think about adding session caching for deletion vector in the next PR.

Reply:
ignore this. it's actually easier to just add a `DeletionLookup` in the session than modify two index